Client/Frontend/Home/ReaderPanel.swift

```diff
@@ -3,51 +3,52 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import UIKit
-
 import Storage
 
-class ReaderPanel: UITableViewController {
-    var profile: Profile!
+class ReaderPanel: UITableViewController, HomePanel {
 
-    private var sites = [Site]()
-    /*
-        Site(title:"Royals Sweep Orioles to Advance to World Series", url:"http://www.nytimes.com/2014/10/16/sports/baseball/royals-keep-rolling-and-advance-to-the-world-series.html?hp&action=click&pgtype=Homepage&version=LargeMediaHeadlineSum&module=photo-spot-region&region=top-news&WT.nav=top-news&_r=0"),
-        Site(title:"How Not to Be Fooled by Odds", url:"http://www.nytimes.com/2014/10/16/upshot/how-not-to-be-fooled-by-odds.html?hp&action=click&pgtype=Homepage&version=HpSum&module=second-column-region&region=top-news&WT.nav=top-news"),
-        Site(title:"Against Rules, Amber Vinson, Dallas Worker With Ebola, Boarded Plane", url:"http://www.nytimes.com/2014/10/16/us/ebola-outbreak-texas.html?hp&action=click&pgtype=Homepage&version=LedeSum&module=first-column-region&region=top-news&WT.nav=top-news")
-    ]
-    */
+    var readingListItemsCursor: Cursor? = nil
+    weak var delegate: HomePanelDelegate? = nil
 
-    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
-        return 1
+    var profile: Profile! {
+        didSet {
+            profile.readingList.get { (cursor) -> Void in
+                self.readingListItemsCursor = (cursor.status == .Success) ? cursor : nil
+                self.tableView.reloadData()
+            }
+        }
     }
 
+    // MARK: UITableViewDataSourceDelegate
+
     override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return sites.count
+        if let cursor = self.readingListItemsCursor {
+            return cursor.count
+        }
+        return 0
```

Comment:
Suggestion: shorten this to <code>return self.readingListItemsCursor?.count ?? 0</code>.